Userland/Applications/PixelPaint/Image.cpp

```diff
@@ -87,6 +80,62 @@ RefPtr<Image> Image::create_from_file(const String& file_path)
     return image;
 }
 
+RefPtr<Image> Image::try_reading_as_bmp(const ByteBuffer& buffer)
+{
+    auto bitmap = Gfx::load_bmp_from_memory(buffer.data(), buffer.size());
+    if (bitmap.is_null())
+        return nullptr;
+
+    return create_from_bitmap(bitmap);
+}
+
+RefPtr<Image> Image::try_reading_as_png(const ByteBuffer& buffer)
+{
+    auto bitmap = Gfx::load_png_from_memory(buffer.data(), buffer.size());
+    if (bitmap.is_null())
+        return nullptr;
+
+    return create_from_bitmap(bitmap);
+}
```

Comment:
This is a job for `Gfx::Bitmap::load_from_file()`. No need for PixelPaint to try different formats.

Reply:
Fixed